vm/rust/src/lib.rs

```diff
@@ -528,16 +530,16 @@ lazy_static! {
 #[allow(static_mut_refs)]
 fn get_versioned_constants(version: *const c_char) -> VersionedConstants {
     let version_str = unsafe { CStr::from_ptr(version) }.to_str().unwrap();
-    let version = StarknetVersion::from_str(&version_str)
-        .unwrap_or(StarknetVersion::from_str(&"0.0.0").unwrap());
+    let version = StarknetVersion::from_str(version_str)
+        .unwrap_or(StarknetVersion::from_str("0.0.0").unwrap());
 
     if let Some(constants) = unsafe { &CUSTOM_VERSIONED_CONSTANTS } {
         constants.clone()
-    } else if version < StarknetVersion::from_str(&"0.13.1").unwrap() {
+    } else if version < StarknetVersion::from_str("0.13.1").unwrap() {
```

Comment:
same here i think a couple of constant fixed values could do. 

or having an UNKNOWN version type 


here is the implementation code block. 
```rust
pub struct StarknetVersion(u8, u8, u8, u8);

...

impl FromStr for StarknetVersion {
    type Err = anyhow::Error;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        if s.is_empty() {
            return Ok(StarknetVersion::new(0, 0, 0, 0));
        }

        let parts: Vec<_> = s.split('.').collect();
        anyhow::ensure!(
            parts.len() == 3 || parts.len() == 4,
            "Invalid version string, expected 3 or 4 parts but got {}",
            parts.len()
        );

        let a = parts[0].parse()?;
        let b = parts[1].parse()?;
        let c = parts[2].parse()?;
        let d = parts.get(3).map(|x| x.parse()).transpose()?.unwrap_or(0);

        Ok(StarknetVersion(a, b, c, d))
    }
}
```

Reply:
I don't think any changes are needed here.
@kirugan What do you think?

Reply:
Agree, let's keep it as is